### PR TITLE
ci: add bug version sync

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref: #46746

### 💡 Background and solution

1. 改进将有严重 bug，不推荐使用的版本信息以 json 方式存储在仓库，方便其他开发者通过 cdn 形式调用。
2. 添加 ci ，将 bug versions 信息同步至 cnpm 仓库。 方便使用 cnpm 的同学 auto-fix

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      --     |
| 🇨🇳 Chinese |      --     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: .github/workflows/sync-bug-versions.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            echo "json<<EOF" >> $GITHUB_OUTPUT
+            cat ./antd/bug-versions.json >> $GITHUB_OUTPUT
+            echo "EOF" >> $GITHUB_OUTPUT
+          fi
+
+      - name: create PR
+        id: cpr
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ secrets.CNPM_BUG_VERSIONS_GITHUB_TOKEN }} # 这里需要 cnpm/bug-versions 的 token 用于推送分支以创建 PR
```
Comment (Wxh16144) [inline .github/workflows/sync-bug-versions.yml:123]: @afc163 这里必须要有一个个人 token。

File: .github/workflows/sync-bug-versions.yml [hunk 2]
```diff
@@ -0,0 +1,144 @@
+# https://github.com/ant-design/ant-design/pull/46746#issuecomment-1873837066
+name: 🔀 Sync bug-versions to cnpm
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "depreciated-versions.json"
+  workflow_dispatch:
```
Comment (Wxh16144) [inline .github/workflows/sync-bug-versions.yml:10]: 手动触发暂时保留，方便 ci 报错时，可以手动再触发一次。

File: depreciated-versions.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{
+  "3.9.3": [
+    "https://github.com/ant-design/ant-design/commit/6550df34b639ab0b3bf2c1cbf9b9828735c1fd41"
+  ],
+  ">= 3.10.0 <=3.10.9": [
+    "https://github.com/ant-design/ant-design/commit/6550df34b639ab0b3bf2c1cbf9b9828735c1fd41"
+  ],
+  ">= 3.11.0 <=3.11.5": [
+    "https://github.com/ant-design/ant-design/commit/6550df34b639ab0b3bf2c1cbf9b9828735c1fd41"
+  ],
```
Comment (Wxh16144) [inline depreciated-versions.json:10]: 因为是直接覆盖 cnpm/bug-versions 的配置，所以把之前的逻辑也 cv 到 antd repo 。（虽然是黑历史...

File: scripts/generate-bug-versions.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+  return head;
+}
+
+async function run() {
+  const bugVersion: IBugVersion = {};
+
+  const { time } = await fetch('http://registry.npmjs.org/antd').then((res) => res.json());
+  // curl -l -o .tmp/antd.json https://registry.npmjs.org/antd // download antd.json
+  // const { time } = (await import('../.tmp/antd.json')).default // local test
```
Comment (Wxh16144) [inline scripts/generate-bug-versions.ts:91]: 为了不频繁拉取 npmjs 信息。本地调试的时候，建议 download 下来测试

File: .github/workflows/sync-bug-versions.yml (hunk 1 above)
Comment (MadCcc) [reply .github/workflows/sync-bug-versions.yml:123]: 放个人 token 不好吧

**Comment by Wxh16144** [comment]
Reason for closure: https://github.com/ant-design/ant-design/pull/46746#issuecomment-1874793712

Reopen #46746

**Pull request closed without merging**